atum/src/main/scala/za/co/absa/atum/utils/S3LocationUtils.scala

```diff
@@ -0,0 +1,42 @@
+package za.co.absa.atum.utils
+
+object S3LocationUtils {
+
+  // hint: https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules
+  val S3LocationRx = "(s3[an]?)://([-a-z0-9.]{3,63})/(.*)".r
+
+  implicit class StringS3LocationExt(path: String) {
+
+    def toS3Location: Option[SimpleS3Location] = {
+      path match {
+        case S3LocationRx(protocol, bucketName, path) => Some(SimpleS3Location(protocol, bucketName, path))
```

Comment:
there are 2 different paths here. Can one be renamed?

Reply:
Thanks, good idea, done.